addons/l10n_ar/models/account_journal.py

```diff
@@ -65,18 +78,16 @@ def _get_journal_letter(self, counterpart_partner=False):
             msg = _('Can not create chart of account until you configure your company AFIP Responsibility and VAT.')
             raise RedirectWarning(msg, action.id, _('Go to Companies'))
 
-        letters = letters_data['issued' if self.type == 'sale' else 'received'][
+        letters = letters_data['issued' if self.l10n_ar_is_pos else 'received'][
             self.company_id.l10n_ar_afip_responsibility_type_id.code]
         if counterpart_partner:
-            counterpart_letters = letters_data['issued' if self.type == 'purchase' else 'received'].get(
+            counterpart_letters = letters_data['issued' if not self.l10n_ar_is_pos else 'received'].get(
                 counterpart_partner.l10n_ar_afip_responsibility_type_id.code, [])
             letters = list(set(letters) & set(counterpart_letters))
         return letters
 
-    def _get_journal_codes(self):
+    def _get_journal_codes_domain(self):
```

Comment:
Why do we have a separate _get_journal_codes_domain and _get_codes_per_journal_type method then?